### DISCUSSION TITLE
Using Raspotify…

### DISCUSSION THREAD

**Discussion by ANGR77** [dtcooper/raspotify#563] [Q&A] [answered]
Hi! 

I am using raspotify to feed a multichannel USB sound card. A Spotify feed uses USB audio port 1 & 2 which in this case is maps to output 1 & 2 on the sound card…but I have  6 more outputs…and it would be interesting to know if it is possible to map output e.g 5 & 6 for raspotify?

**Comment by JasonLG1979** [comment, marked as answer]
You can use ALSA to split and duplicate the channels as many times as you like, but it's not for the faint of heart. It will require a pretty complex `asound.conf` and a lot of trial and error, and depending on the card you may have audio syncing issues. The timing can drift which could cause weird phase cancellation effects. You're welcome to give it a shot though.

Here are the ALSA PCM plugin docs:

https://vovkos.github.io/doxyrest/samples/alsa-sphinxdoc/page_pcm_plugins.html

**Reply by JasonLG1979** [reply]
This might also be of some help?

https://alsa.opensrc.org/SurroundSound

**Reply by ANGR77** [reply]
Big thanks! 
I will make an odyssey in to this and see if I can map the two stereo channels to other ports than usb 1&2. On my sound device - e.g a digital mixer with usb which virtually can receive up to 18 usb input ports- and it would be very nice to be able to send raspotify streams directly to for an example on USB port 17 & 18. (It is a Behringer XR18 mixer with 18 inputs and 8 outputs…) 

best regards

Anders

**Reply by JasonLG1979** [reply]
You can use lowpass/hipass filters, combine channels and introduce delays as well to turn stereo audio into a sort of virtual surround sound.

**Reply by JasonLG1979** [reply]
@ANGR77 You might try [The Audio Config Wizard](https://github.com/dtcooper/raspotify/wiki/Basic-Setup-Guide#the-easy-way) I've been playing with if you haven't figured it out for yourself yet. I recently add Up/Down channel mixing support.

Some cards don't map their channels correctly so you may need to manually edit the generated `/etc/asound.conf` but it should get you most of the way there at the very least.
